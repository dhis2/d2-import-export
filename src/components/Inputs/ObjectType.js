import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { SelectField } from '../../components/'
import { objectTypeOptions } from '../../utils/options'

const NAME = 'objectType'
const LABEL = i18n.t('Object type')
const DATATEST = 'input-object-type'

const ObjectType = () => (
    <SelectField
        name={NAME}
        label={LABEL}
        options={objectTypeOptions}
        filled
        initialFocus
        dataTest={DATATEST}
    />
)

export { ObjectType }
