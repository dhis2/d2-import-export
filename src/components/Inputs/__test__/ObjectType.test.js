import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { ObjectType } from '../ObjectType'

jest.mock('react-final-form')

describe('Input component - ObjectType', () => {
    it('should render correctly', () => {
        const file = shallow(<ObjectType />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
