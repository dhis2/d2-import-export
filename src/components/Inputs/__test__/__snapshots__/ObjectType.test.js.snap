// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - ObjectType should render correctly 1`] = `
<Field>
  <Select
    dataTest="input-object-type"
    label="Object type"
    name="objectType"
    options={
      Array [
        Object {
          "label": "Data sets",
          "value": "dataSets",
        },
        Object {
          "label": "Programs",
          "value": "programs",
        },
        Object {
          "label": "Category combination",
          "value": "categoryCombos",
        },
        Object {
          "label": "Dashboard",
          "value": "dashboards",
        },
        Object {
          "label": "Data element groups",
          "value": "dataElementGroups",
        },
        Object {
          "label": "Option sets",
          "value": "optionSets",
        },
      ]
    }
  />
</Field>
`;
